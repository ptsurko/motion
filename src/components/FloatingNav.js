import React, { PureComponent } from 'react';
import Types from 'prop-types';
import ConsoleSvg from '../resources/ic_code.svg';
import ShareSvg from '../resources/Share.svg';
import { ShareContainer } from '@satori-sdk/component-library';

class FloatingNav extends PureComponent {
  render() {
    const { onConsoleClick, participantsCount, showConsole, showParticipants } = this.props;

    return (
      <section className="floatingNav">
        {
          showParticipants &&
          <section className="floatingNav-item floatingNav-devices">
            <div className="participantsCount">{participantsCount}</div>
            <div>Devices</div>
          </section>
        }
        <div className="floatingNav-actionItems">
          {
            showConsole &&
            <section
              className="floatingNav-item floatingNav-console"
              onClick={onConsoleClick}
            >
              <img
                className="share-img"
                src={ConsoleSvg}
                alt="Console"
              />
              <div>Inspect</div>
            </section>
          }
          <section ref="share" className="floatingNav-item floatingNav-share">
            <ShareContainer
              className="shareTrigger"
              placement="left"
            >
              <img
                className="share-img"
                src={ShareSvg}
                alt="Share URL"
              />
              <div>Share</div>
            </ShareContainer>
          </section>
        </div>
      </section>
    );
  }
}

FloatingNav.propTypes = {
  showConsole: Types.bool.isRequired,
  showParticipants: Types.bool.isRequired,
  participantsCount: Types.number,
};

export default FloatingNav;
