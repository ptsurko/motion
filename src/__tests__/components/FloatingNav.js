import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import FloatingNav from 'components/FloatingNav';

it('renders FloatingNav component correctly when showParticipants is false', () => {
  const Component = <FloatingNav showConsole={false} showParticipants={false} />;
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();

  const wrapper = shallow(Component);
  expect(wrapper.find('.floatingNav-devices').length).toBe(0);
  expect(wrapper.find('.floatingNav-share').length).toBe(1);
});

it('renders FloatingNav component correctly when showParticipants is true', () => {
  const Component = <FloatingNav showConsole={false} showParticipants={true} participantsCount={5} />;
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();

  const wrapper = shallow(Component);
  expect(wrapper.find('.floatingNav-devices').length).toBe(1);
  expect(wrapper.find('.participantsCount').text()).toBe('5');
  expect(wrapper.find('.floatingNav-share').length).toBe(1);
});

it('should respond to clicking on the share button', () => {
  const Component = <FloatingNav showConsole={false} showParticipants={true} participantsCount={5} />;
  const wrapper = mount(Component);
  const shareTrigger = wrapper.find('.shareWidget-container');

  expect(document.querySelector('.shareWidget')).toBeNull();

  shareTrigger.simulate('click');
  expect(document.querySelector('.shareWidget')).not.toBeNull();

  shareTrigger.simulate('click');
  expect(document.querySelector('.shareWidget')).toBeNull();
});
