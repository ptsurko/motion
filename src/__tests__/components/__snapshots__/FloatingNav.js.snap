// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FloatingNav component correctly when showParticipants is false 1`] = `
<section
  className="floatingNav"
>
  <div
    className="floatingNav-actionItems"
  >
    <section
      className="floatingNav-item floatingNav-share"
    >
      <div
        className="shareWidget-container shareTrigger"
        onClick={[Function]}
      >
        <img
          alt="Share URL"
          className="share-img"
          src="Share.svg"
        />
        <div>
          Share
        </div>
      </div>
    </section>
  </div>
</section>
`;

exports[`renders FloatingNav component correctly when showParticipants is true 1`] = `
<section
  className="floatingNav"
>
  <section
    className="floatingNav-item floatingNav-devices"
  >
    <div
      className="participantsCount"
    >
      5
    </div>
    <div>
      Devices
    </div>
  </section>
  <div
    className="floatingNav-actionItems"
  >
    <section
      className="floatingNav-item floatingNav-share"
    >
      <div
        className="shareWidget-container shareTrigger"
        onClick={[Function]}
      >
        <img
          alt="Share URL"
          className="share-img"
          src="Share.svg"
        />
        <div>
          Share
        </div>
      </div>
    </section>
  </div>
</section>
`;
